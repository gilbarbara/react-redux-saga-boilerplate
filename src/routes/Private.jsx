import React from 'react';
import styled from 'styled-components';

import Github from 'containers/GitHub';

import { Box, Container, Heading, Link, Paragraph, Screen, Text, utils } from 'styled-minimal';

const Header = styled.div`
  margin-bottom: ${utils.spacer(3)};
  text-align: center;
`;

const PrivateWrapper = styled(Screen)``;

const Private = () => (
  <PrivateWrapper key="Private">
    <Container verticalPadding>
      <Header>
        <Heading>Oh hai!</Heading>
        <Paragraph>
          You can get this boilerplate{' '}
          <Link href="https://github.com/gilbarbara/react-redux-saga-boilerplate/" target="_blank">
            here
          </Link>
        </Paragraph>
      </Header>
      <Box textAlign="center" mb={4}>
        <Heading as="h5">Here's some GitHub data</Heading>
        <Text fontSize={1}>
          <i>*Just to have some requests in the sagas...</i>
        </Text>
      </Box>
      <Github />
    </Container>
  </PrivateWrapper>
);

export default Private;
