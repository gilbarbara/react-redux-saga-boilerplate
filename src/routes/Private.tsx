import { Anchor, Box, H5, Jumbo, Page, Paragraph, Text } from '@gilbarbara/components';

import Github from '~/containers/GitHub';

function Private() {
  return (
    <Page key="Private" data-component-name="Private">
      <Box mb="lg" textAlign="center">
        <Jumbo mb={0}>Oh hai!</Jumbo>
        <Paragraph>
          You can get this boilerplate{' '}
          <Anchor external href="https://github.com/gilbarbara/react-redux-saga-boilerplate/">
            here
          </Anchor>
        </Paragraph>
      </Box>
      <Box mb="xl" textAlign="center">
        <H5 mb={0}>Here's some GitHub data</H5>
        <Text size="xs">
          <i>*Just to have some requests in the sagas...</i>
        </Text>
      </Box>
      <Github />
    </Page>
  );
}

export default Private;
