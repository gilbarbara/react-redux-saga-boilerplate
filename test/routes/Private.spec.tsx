import React from 'react';
import { render, screen, waitForElementToBeRemoved } from 'test-utils';

import Private from '~/routes/Private';

describe('Private', () => {
  it('should render properly', async () => {
    render(<Private />);

    expect(screen.getByTestId('Private')).toMatchSnapshot();

    await waitForElementToBeRemoved(() => screen.queryByTestId('LoaderPill'));
  });
});
