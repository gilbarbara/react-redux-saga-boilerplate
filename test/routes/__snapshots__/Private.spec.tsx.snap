// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Private > should render properly 1`] = `
<div
  class="css-px2j5d"
  data-component-name="Private"
>
  <div
    class="css-wc6egs"
    data-component-name="Box"
  >
    <div
      class="css-1tpc2wv"
      data-component-name="Box"
    >
      <h1
        class="css-hnj43m"
        data-component-name="Jumbo"
      >
        Oh hai!
      </h1>
      <p
        class="css-15bhhaq"
        data-component-name="Paragraph"
      >
        <span
          class="css-lo0v0o"
          data-component-name="Text"
        >
          You can get this boilerplate
           
          <a
            class="css-15oyb1u"
            data-component-name="Anchor"
            href="https://github.com/gilbarbara/react-redux-saga-boilerplate/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span>
              here
            </span>
          </a>
        </span>
      </p>
    </div>
    <div
      class="css-l3usq0"
      data-component-name="Box"
    >
      <h5
        class="css-wqaawd"
        data-component-name="H5"
      >
        Here's some GitHub data
      </h5>
      <span
        class="css-bj43zy"
        data-component-name="Text"
      >
        <i>
          *Just to have some requests in the sagas...
        </i>
      </span>
    </div>
    <div
      data-component-name="GitHubWrapper"
    >
      <div
        class="css-1pedvdu"
        data-component-name="Box"
      >
        <div
          aria-label="GitHub Selector"
          class="css-179edw6"
          data-component-name="GitHubSelector"
          role="group"
        >
          <button
            class="css-hepdsj"
            data-component-name="Button"
            data-value="react"
            type="button"
          >
            React
            <span
              class="css-1sqvwf5"
              data-component-name="Icon"
              name="spinner"
            >
              <svg
                height="20"
                src="https://cdn.jsdelivr.net/npm/css.gg/icons/svg/spinner.svg"
                title="spinner"
                width="20"
              />
            </span>
          </button>
          <button
            class="css-1umnmkm"
            data-component-name="Button"
            data-value="redux"
            type="button"
          >
            Redux
          </button>
        </div>
      </div>
      <div
        class="css-1gnxr2w"
        data-component-name="LoaderPill"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
